### PR TITLE
Add a new package: unblur.

### PR DESCRIPTION
**Pull request by weijianwen**

### CODE REVIEW

File: var/spack/repos/builtin/packages/unblur/package.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    variant('shared', default=True, description='Dynamic linking')
+
+    depends_on('zlib')
+    depends_on('jpeg')
+    depends_on('libtiff')
+    depends_on('gsl')
+    depends_on('jbigkit')
+    depends_on('fftw@3:')
+    # Requires Intel Fortran compiler
+    #depends_on('%intel')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/unblur/package.py]: If you want to enforce that the Intel compilers are required, you can add `conflicts()` directives for every other compiler.

File: var/spack/repos/builtin/packages/unblur/package.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+from spack import *
+
+
+class Unblur(Package):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/unblur/package.py]: You should use `AutotoolsPackage`.

File: var/spack/repos/builtin/packages/unblur/package.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    depends_on('jpeg')
+    depends_on('libtiff')
+    depends_on('gsl')
+    depends_on('jbigkit')
+    depends_on('fftw@3:')
+    # Requires Intel Fortran compiler
+    #depends_on('%intel')
+
+    def install(self, spec, prefix):
+        with working_dir('src'):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/unblur/package.py]: Add:
```python
configure_directory = 'src'
```
up above.

File: var/spack/repos/builtin/packages/unblur/package.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            filter_file(r'<<<<<<<.*', '', 'missing')
+            configure('FC=ifort',
+                      'F77=ifort',
+                      '--prefix={0}'.format(prefix),
+                      '--enable-static={0}'.format('yes' if '~shared' in spec else 'no'),
+                      '--enable-openmp={0}'.format('yes' if '+openmp' in spec else 'no'),
+                      '--enable-optimisations=yes')
+            make(parallel=False)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/unblur/package.py]: Replace this with:
```python
def build(self, spec, prefix):
    with working_dir(self.build_directory):
        make(parallel=False)
```

File: var/spack/repos/builtin/packages/unblur/package.py (hunk 2 above)
Comment (weijianwen) [reply var/spack/repos/builtin/packages/unblur/package.py]: Fixed.

File: var/spack/repos/builtin/packages/unblur/package.py (hunk 3 above)
Comment (weijianwen) [reply var/spack/repos/builtin/packages/unblur/package.py]: Fixed.

File: var/spack/repos/builtin/packages/unblur/package.py (hunk 4 above)
Comment (weijianwen) [reply var/spack/repos/builtin/packages/unblur/package.py]: Fixed.

File: var/spack/repos/builtin/packages/unblur/package.py [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+    depends_on('zlib')
+    depends_on('jpeg')
+    depends_on('libtiff')
+    depends_on('gsl')
+    depends_on('jbigkit')
+    depends_on('fftw@3:')
+    # Requires Intel Fortran compiler
+    conflicts('%gcc')
+    conflicts('%pgi')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/unblur/package.py]: There are a few other compilers you need to add as conflicts:
```
$ ls lib/spack/spack/compilers/
__init__.py  cce.pyc      gcc.pyc      nag.pyc      xl.pyc
__init__.pyc clang.py     intel.py     pgi.py       xl_r.py
__pycache__  clang.pyc    intel.pyc    pgi.pyc      xl_r.pyc
cce.py       gcc.py       nag.py       xl.py
```

Comment (weijianwen) [reply var/spack/repos/builtin/packages/unblur/package.py]: Fixed.

**Pull request merged**